### ISSUE TITLE
add csv output option for `lca rankinfo`

### ISSUE THREAD

**Issue by bluegenes** [sourmash-bio/sourmash#1920]
`lca rankinfo` is really informative for looking at unique k-mers at different taxonomic levels, but the output is more human-friendly than computer-friendly. Would be great to optionally output a csv for easy aggregation of `lca rankinfo` runs on multiple databases.

**Comment by ctb** [comment]
👀 I mean, sure? It should be an easy PR, Tessa. 😆

**Comment by bluegenes** [comment]
Yeah yeah, it's probably something I'll throw in shortly :). Hand copying/reformatting is not my jam, and I wants to run additional databases!

While I'm at it, I think it might also be pretty simple to enable a `--singleton` or `--by-database` mode to allow running multiple databases separately, rather than in aggregate (currently counts k-mers from all input databases together). Downside: not parallel, would take longer than running one job per database. Upside: simpler from user point of view.

**Comment by ctb** [comment]
> Yeah yeah, it's probably something I'll throw in shortly :). Hand copying/reformatting is not my jam, and I wants to run additional databases!

👍
 
> While I'm at it, I think it might also be pretty simple to enable a `--singleton` or `--by-database` mode to allow running multiple databases separately, rather than in aggregate (currently counts k-mers from all input databases together). Downside: not parallel, would take longer than running one job per database. Upside: simpler from user point of view.

hot take: there are no real advantages on the computational side to supporting `--singleton` style execution here. Loading will take as long as it takes, and will not be sped up. So `--singleton` or `--by-database` is "just" a UX issue.

(The reverse is not true: if we _didn't_ support merged functionality as it is at present, you wouldn't be able to merge across multiple databases!)

Also, it's interesting how the LCA database functionality, which we don't like for taxonomy purposes any more, remains useful for other things! 🤔
